examples/parsing/explicit_curry.an

```diff
@@ -0,0 +1,17 @@
+
+foo _ a
+oof 1 _ 3
+
+bar _ "test" _
+
+baz 1 (_, 32)
+qux 3 (_) _
+
+
+// args: --parse
+// expected stdout:
+// (\ _$1 . (foo _$1 a));
```

Comment:
Ah, we should probably update the ast printout syntax for lambdas in a future PR to match the new `fn params -> body` syntax